src/streaming/streamingclient.py

```diff
@@ -0,0 +1,144 @@
+"""StreamingClient tool used for streaming services"""
+from src.streaming.MediaConfig import MediaConfig
+import websocket
+import io
+import threading
+import six
+import json
+
+def on_error(error):
+    raise error
+
+def on_close(code, reason):
+    print("Connection Closed. Code : {}; Reason : {}".format(code, reason))
+
+def on_connected(job_id):
+    print('Connected, Job ID : {}'.format(job_id))
+
+class RevAiStreamingClient():
+    def __init__(self, 
+                access_token, 
+                config, 
+                version = 'v1alpha',  
+                on_error = on_error, 
+                on_close = on_close, 
+                on_connected = on_connected):
+        """Constructor for Streaming Client
+
+        :param access_token: access token which authorizes all requests and 
+            links them to your account. Generated on the settings page of your 
+            account dashboard on Rev.ai.
+        :param config: an MediaConfig object containing audio information. 
+            See MediaConfig.py for more information
+        :param version (optional): version of the streaming speech to text as  
+            string
+        :param on_message (optional): function to be called when recieving a 
+            message from the server
+        :param on_error (optional): function to be called when recieving an 
+            error from the server
+        :param on_close (optional): function to be called when the websocket 
+            closes
+        :param on_connected (optional): function to be called when the websocket
+            and thread starts successfully
+        """
+        if not access_token:
+            raise ValueError('access_token must be provided')
+
+        if not config:
+            raise ValueError('config must be provided')
+
+        self.access_token = access_token
+        self.config = config
+        self.base_url = base_url = 'wss://api.rev.ai/speechtotext/{}/stream'. \
```

Comment:
consider using f-strings if on python3.6+ it makes formatting a lot easier. https://docs.python.org/3/whatsnew/3.6.html#whatsnew36-pep498

Reply:
We want to support python 2.7 and 3.4+ so I don't think we can use these